service/models.py

```diff
@@ -63,22 +64,56 @@ class Shopcart(db.Model):
     ##################################################
     id = db.Column(db.Integer, primary_key=True)
     user_id = db.Column(db.Integer)
-    create_time = db.Column(db.DateTime)
-    update_time = db.Column(db.DateTime)
+    create_time = db.Column(db.DateTime, server_default=func.now())
+    update_time = db.Column(db.DateTime, server_default=func.now(), onupdate=func.now())
 
     ##################################################
     # INSTANCE METHODS
     ##################################################
 
+    def __repr__(self):
+        return "<Shopcart %r>" % (self.id)
+
+    def create(self):
+        """
+        Creates a Shopcart to the database
+        """
+        date_time = datetime.now()
+        self.id = None  # id must be none to generate next primary key
+        #if time fields not set, use current time
+        if self.create_time is None:
+            self.create_time = date_time
+        if self.update_time is None:
+            self.update_time = date_time
```

Comment:
Just curious, if we already have `server_default` and `onupdate`, do we need to set the value here?

Reply:
Yeah it wasn't working for me, I thought maybe in future could go back and try and fix this, but didn't want to hold everything up.